Mathlib/Algebra/Group/Equiv/Basic.lean

```diff
@@ -75,8 +75,7 @@ structure AddEquiv (A B : Type*) [Add A] [Add B] extends A ≃ B, AddHom A B
 
 /-- `AddEquivClass F A B` states that `F` is a type of addition-preserving morphisms.
 You should extend this class when you extend `AddEquiv`. -/
-class AddEquivClass (F : Type*) (A B : outParam (Type*)) [Add A] [Add B]
-    extends EquivLike F A B where
+class AddEquivClass (F A B : Type*) [Add A] [Add B] [EquivLike F A B] : Prop where
```

Comment:
I am confused at to whether one should use `outParam`s or not: here you have removed outParam, but in many other classes you have kept them. Can you explain if there is a rule of thumb to be followed?

Reply:
The rule of thumb is to keep `outParam` even if it's not strictly necessary: this seems to help with unification problems, for example in the category theory library. However I didn't know this at the start of the PR so I removed `outParam` at first. I might have missed other places where I removed it and didn't add it back.